api-server/src/events/events.service.ts

```diff
@@ -47,14 +47,47 @@ export class EventsService {
         return this.eventModel.findAll(findOptions)
     }
 
-    update(id: string, values: Partial<UpdateEventRequest>): Promise<DbUpdateResponse<EventModel>> {
-        const updateQueryOptions: UpdateOptions = {
-            where: {id},
-            returning: true
-        }
+    async update(id: string, values: Partial<UpdateEventRequest>): Promise<DbUpdateResponse<EventModel>> {
+
+        return this.sequelize.transaction(async (transaction) => {
+            const transactionHost = { transaction };
+            const updateQueryOptions: UpdateOptions = {
+                where: {id},
+                returning: true,
+                transaction
+            }
+
+            const updatedEvent = await this.eventModel.update(values, updateQueryOptions)
+
+            await this.updateDatetimeVenueEntries(
+                id,
+                values.venue_id,
+                values.date_times,
+                transactionHost);
+
+            return updatedEvent;
```

Comment:
This probably isn't important because I don't think the frontend uses this for anything, but I wonder if `updatedEvent` will have the old datetimes?

Reply:
currently the frontend may not, but I think it's a good pattern to always return the updated state. It allows the client to know about any autogenerated fields such ids or timestamps without having to refetch.

If we wanted to be complete in our use of the pattern we should probably create a return type that allows us to mix back in the final state of the DateTimes.

It could be something like

```
EventUpdateResponse {
   eventModel: EventModel,
   date_times: DatetimeVenueModel [];
}
```

In any event, like Mathew says, the frontend doesn't currently do anything with this; so, not that important. I'm mostly pointing it out to explain the general pattern I've been trying to follow with endpoints.

Feel free to tweak it some more or not. There will always be more to refine :-)